iree/turbine/kernel/wave/codegen.py

```diff
@@ -1076,6 +1077,26 @@ def handle_shared_memory_barrier(emitter: WaveEmitter, node: fx.Node):
 ###############################################################################
 
 
+@handle_op(extract)
+def handle_extract(emitter: WaveEmitter, node: fx.Node):
+    try:
+        register, offset = node.args
+    except ValueError as e:
+        raise ValidationError("Malformed arguments") from e
+    assert isinstance(offset, list) and len(offset) == 1
+    extract_vector = cast_vector(emitter, register)
+    result_type = VectorType.get([1], extract_vector.type.element_type)
+    element = vector_d.extract_strided_slice(
```

Comment:
Just wondering why an extract() wouldn't work here instead?

Reply:
it does and that's what we should do, it was just easier to use the previous implementation haha

Reply:
Tried moving it to vector_d.extract, but actually it makes our shuffleOp lowering a little more complicated since not all inputs are vectorType. Additionally, it may be good to keep extract_strided_slice for future upgrade to multi-row extract.